lib/services/role.go

```diff
@@ -345,11 +345,23 @@ func ApplyTraits(r types.Role, traits map[string][]string) types.Role {
 			r.SetDatabaseLabels(condition, applyLabelsTraits(inLabels, traits))
 		}
 
+		inHostGroups := r.GetHostGroups(condition)
+		var outHostGroups []string
+		for _, group := range inHostGroups {
+			vals, err := ApplyValueTraits(group, traits)
+			if err != nil && !trace.IsNotFound(err) {
+				log.Warnf("Did not apply trait to host group: %v", err)
+				continue
+			}
+			outHostGroups = append(outHostGroups, vals...)
+		}
```

Comment:
You can simplify this and just use `applyValueTraitsSlice` function, see examples above.